apiserver/test/integration/clientset_test.go

```diff
@@ -1349,3 +1355,109 @@ func testIPAMConfigClient(client calicoclient.Interface, name string) error {
 
 	return nil
 }
+
+// TestBlockAffinityClient exercises the BlockAffinity client.
+func TestBlockAffinityClient(t *testing.T) {
+	const name = "test-blockaffinity"
+	rootTestFunc := func() func(t *testing.T) {
+		return func(t *testing.T) {
+			client, shutdownServer := getFreshApiserverAndClient(t, func() runtime.Object {
+				return &v3.BlockAffinity{}
+			})
+			defer shutdownServer()
+			if err := testBlockAffinityClient(client, name); err != nil {
+				t.Fatal(err)
+			}
+		}
+	}
+
+	if !t.Run(name, rootTestFunc()) {
+		t.Errorf("test-blockaffinity test failed")
+	}
+}
+
+func testBlockAffinityClient(client calicoclient.Interface, name string) error {
+	blockAffinityClient := client.ProjectcalicoV3().BlockAffinities()
+	blockAffinity := &v3.BlockAffinity{
+		ObjectMeta: metav1.ObjectMeta{Name: name},
+
+		Spec: v3.BlockAffinitySpec{
+			CIDR:  "10.0.0.0/24",
+			Node:  "node1",
+			State: "pending",
+		},
+	}
+	libV3BlockAffinity := &libapiv3.BlockAffinity{
+		ObjectMeta: metav1.ObjectMeta{Name: name},
+
+		Spec: libapiv3.BlockAffinitySpec{
+			CIDR:    "10.0.0.0/24",
+			Node:    "node1",
+			State:   "pending",
+			Deleted: "false",
+		},
+	}
+	ctx := context.Background()
+
+	// Calico libv3 client instantiation in order to get around the API create restrictions
+	// TODO: Currently these tests only run on a Kubernetes datastore since profile creation
```

Comment:
"profile"

Reply:
I meant to use the "profile" here since profiles are the actual reason that we only run KDD tests and not etcd tests. It's a tech debt piece we haven't really addressed yet. I can remove mentioning profiles if it's confusing but then might not be as clear when these eventually start failing due to the profile issue being cleaned up.

Reply:
ah ok